maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporter.java

```diff
@@ -132,9 +133,27 @@ public void testSetCompleted( WrappedReportEntry testSetReportEntry, TestSetStat
             createTestSuiteElement( ppw, testSetReportEntry, testSetStats, testSetReportEntry.elapsedTimeAsString() );
 
             showProperties( ppw, testSetReportEntry.getSystemProperties() );
-
+            
+            
+
+            //map.entrySet() Convert into list
+            List<Map.Entry<String, List<WrappedReportEntry>>> methodRunHistoryMapCoIntoList =
+                    new ArrayList<Map.Entry<String, List<WrappedReportEntry>>>( methodRunHistoryMap.entrySet() );
+            //sort
```

Comment:
pls do not use comments.
Can you use static import instead of this long line `Collections.sort`? And import `Map.Entry` as well. Then pls do not rephrase reference type in local variable name `methodRunHistoryMapCoIntoList` and use plural of `methodReportEntries`.
Can you please extract this part into possibly a private static method and write a unit test just for that? You should call private method with PowerMock using `org.powermock.reflect.Whitebox;`. As a hint see our unit tests in Surefire.